iree/turbine/kernel/wave/templates/paged_decode_attention.py

```diff
@@ -257,19 +261,20 @@ def phase_0(
         # Output has shape [NUM_KV_SPLITS, NUM_SEQS, NUM_HEADS, HEAD_DIM]
         # =========================================================================
 
-        layer_scale_reg = tkl.Register[S, B, K2, tkl.f32](layer_scaling)
+        layer_scale_reg = tkl.Register[S, B, K2_ITER, tkl.f32](layer_scaling)
 
         init_max = tkl.Register[S, B, tkl.f32](-1e6)
         init_sum = tkl.Register[S, B, tkl.f32](0.0)
         new_acc = tkl.Register[S, N, B, tkl.f32](0.0)
 
-        zero = tkl.Register[B, K2, tkl.f32](0.0)
-        neg_infinity = tkl.Register[B, K2, tkl.f32](-1e6)
+        zero = tkl.Register[B, K2_ITER, tkl.f32](0.0)
+        neg_infinity = tkl.Register[B, K2_ITER, tkl.f32](-1e6)
 
         # The request index is used to load the appropriate entries from the block table.
         req_index = tkw.read(request_indices)
         # The sequence length is used to control the bounds of the loop over K2.
         seq_length = tkw.read(request_indices, mapping=seq_len_mapping)
+        tkw.set_symbol(K2_ITER, seq_length)
```

Comment:
Can do in future changes, but we should add a comment saying that seq_length depends on the batch/sequenceIdx => different WG will have different K2 size